.drone.yml

```diff
@@ -2584,6 +2584,27 @@ steps:
     GITHUB_PRIVATE_KEY:
       from_secret: GITHUB_PRIVATE_KEY
     WORKSPACE_DIR: /tmp/build-darwin-amd64
+- name: Install Go Toolchain
+  commands:
+  - set -u
+  - mkdir -p ~/build-$DRONE_BUILD_NUMBER-$DRONE_BUILD_CREATED-toolchains
+  - curl --silent -O https://dl.google.com/go/$RUNTIME.darwin-amd64.tar.gz
+  - tar -C  ~/build-$DRONE_BUILD_NUMBER-$DRONE_BUILD_CREATED-toolchains -xzf $RUNTIME.darwin-amd64.tar.gz
+  - rm -rf $RUNTIME.darwin-amd64.tar.gz
+  environment:
+    RUNTIME: go1.17.2
+- name: Install Rust Toolchain
+  commands:
+  - set -u
+  - export PATH=/Users/build/.cargo/bin:$PATH
+  - mkdir -p ~/build-$DRONE_BUILD_NUMBER-$DRONE_BUILD_CREATED-toolchains
+  - export RUST_VERSION=$(grep RUST_VERSION $WORKSPACE_DIR/go/src/github.com/gravitational/teleport/build.assets/Dockerfile
+    | cut -d= -f2)
+  - export CARGO_HOME=~/build-$DRONE_BUILD_NUMBER-$DRONE_BUILD_CREATED-toolchains
+  - export RUST_HOME=$CARGO_HOME
+  - rustup toolchain install $RUST_VERSION
+  environment:
+    WORKSPACE_DIR: /tmp/build-darwin-amd64
 - name: Build Mac release artifacts
```

Comment:
I think you'll have to update PATH for the build artifacts step so that the per-build tool chains are detected first.